src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -981,6 +975,35 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         }
     }
 
+    // Method Description:
+    // - Pre-process (if necessary) text pasted (presumably from the clipboard)
+    //   before sending it over the terminal's connection, respecting
+    //   appropriate terminal settings
+    void TermControl::_SendPastedTextToConnection(const std::wstring& wstr)
+    {
+        // Check settings to see if we should be stripping
+        // out line feeds
+        if (_settings.StripLineFeedsOnPaste())
+        {
+            std::wstring stripped(wstr);
+
+            std::wstring::size_type pos = 0;
+
+            // Lament that the stl string does not have
+            // a search/replace method
+            while ((pos = stripped.find(L'\r', pos)) != std::wstring::npos)
+            {
+                stripped.replace(pos, 1, L"");
+            }

```

Comment:
Maybe consider std::regex_replace? Something like
std::regex_replace(stripped, std::wregex(L"(\r\n?|\n)"), L"");

I am not sure what the line endings would need to be considered since this may come from the clipboard but a correct regex could handle them. 

[This](https://regex101.com/) is nice for testing the regex. 

Since I don't know the details of this - your the find/replace code may cover all of the cases.  Just thinking out loud.

Reply:
duh - I completely misunderstood what was going on here.  The regex is wrong here.  The idea is the same though.  I'm not sure it is that much better though.  It is shorter though. 😄

Reply:
@dlong11 the regex may still be the better way to do it, but in thinking about this, I've come to two realizations:

1. This should just be replacing Windows-space line endings with Unix-space line endings
2. I have had which character is _named_ which in the CRLF pair backwards for my entire career and this is the first time having the name wrong has actually mattered -- the code as written strips carriage returns, not line feeds.

Reply:
@d-bingham Funny how that happens.  I know I have a few things like that too!   The regex should be easy to modify to replace CRLF with LF then.   Again I am unsure if unicode line feeds are an issue.  You may be able to user \R and replace with \r (LF)  Not sure.   I always have to look in the docs to get regex's to work.   So I shouldn't be giving advice on them.  🤦‍♂

Reply:
@dlong11 updated the search/replace, and to my eye, at least, it's easier to read than the regex.

Honestly I think the core issue with line endings is that Terminal inserts ASCII CRLF pairs into text copied to the clipboard, which WSL's bash promptly handles poorly.  Fixing it on the copy side has side effects when pasting text copied from the terminal to Windows-space programs, and fixing it on the paste side has the advantage of automatically fixing pastes from other Windows-space stuff that also generates CRLF pairs.